File: samples/aspire-with-javascript/ts/apphost.ts
import { createBuilder } from './.modules/aspire.js';

const builder = await createBuilder();

const weatherApi = await builder.addProject("weatherapi", "../AspireJavaScript.MinimalApi/AspireJavaScript.MinimalApi.csproj", "https")
    .withExternalHttpEndpoints();

await builder.addJavaScriptApp("angular", "../AspireJavaScript.Angular", { runScriptName: "start" })
    .withReference(weatherApi)
    .waitFor(weatherApi)
    .withHttpEndpoint({ env: "PORT" })
    .withExternalHttpEndpoints()
    .publishAsDockerFile();

await builder.addJavaScriptApp("react", "../AspireJavaScript.React", { runScriptName: "start" })
    .withReference(weatherApi)
    .waitFor(weatherApi)
    .withEnvironment("BROWSER", "none")
    .withHttpEndpoint({ env: "PORT" })
    .withExternalHttpEndpoints()
    .publishAsDockerFile();

await builder.addJavaScriptApp("vue", "../AspireJavaScript.Vue", { runScriptName: "start" })
    .withReference(weatherApi)
    .waitFor(weatherApi)
    .withHttpEndpoint({ env: "PORT" })
    .withExternalHttpEndpoints()
    .publishAsDockerFile();

const reactVite = await builder.addViteApp("reactvite", "../AspireJavaScript.Vite")
    .withReference(weatherApi)
    .withEnvironment("BROWSER", "none");

await weatherApi.publishWithContainerFiles(reactVite, "./wwwroot");

await builder.build().run();

File: samples/aspire-with-javascript/AspireJavaScript.AppHost/AppHost.cs
#pragma warning disable ASPIREBROWSERLOGS001
#pragma warning disable ASPIREJAVASCRIPT001

var builder = DistributedApplication.CreateBuilder(args);

var weatherApi = builder.AddProject<Projects.AspireJavaScript_MinimalApi>("weatherapi")
    .WithExternalHttpEndpoints();

builder.AddJavaScriptApp("angular", "../AspireJavaScript.Angular", runScriptName: "start")
    .WithNpm(installCommand: "ci")
    .WithReference(weatherApi)
    .WaitFor(weatherApi)
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints()
    .WithBrowserLogs()
    .PublishAsStaticWebsite("/api", weatherApi, options => options.OutputPath = "dist/weather/browser");

builder.AddJavaScriptApp("react", "../AspireJavaScript.React", runScriptName: "start")
    .WithNpm(installCommand: "ci")
    .WithReference(weatherApi)
    .WaitFor(weatherApi)
    .WithEnvironment("BROWSER", "none") // Disable opening browser on npm start
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints()
    .WithBrowserLogs()
    .PublishAsStaticWebsite("/api", weatherApi);

builder.AddJavaScriptApp("vue", "../AspireJavaScript.Vue")
    .WithRunScript("start")
    .WithNpm(installCommand: "ci")
    .WithReference(weatherApi)
    .WaitFor(weatherApi)
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints()
    .WithBrowserLogs()
    .PublishAsStaticWebsite("/api", weatherApi);

builder.AddViteApp("reactvite", "../AspireJavaScript.Vite")
    .WithNpm(installCommand: "ci")
    .WithReference(weatherApi)
    .WithEnvironment("BROWSER", "none")
    .WithBrowserLogs()
    .PublishAsStaticWebsite("/api", weatherApi);

builder.Build().Run();
